staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/clock/event_clock_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clock
+
+import (
+	"math/rand"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"k8s.io/utils/clock"
+)
+
+type TestableEventClock interface {
+	EventClock
+	SetTime(time.Time)
+	Run(*time.Time)
+}
+
+func exerciseTestableEventClock(t *testing.T, ec TestableEventClock, fuzz time.Duration) {
+	exercisePassiveClock(t, ec)
+	var numDone int32
+	now := ec.Now()
+	strictable := true
+	const batchSize = 100
+	times := make(chan time.Time, batchSize+1)
+	try := func(abs, strict bool, d time.Duration) {
+		f := func(u time.Time) {
+			realD := ec.Since(now)
+			atomic.AddInt32(&numDone, 1)
+			times <- u
+			if realD < d || strict && strictable && realD > d+fuzz {
+				t.Errorf("Asked for %v, got %v", d, realD)
+			}
+		}
+		if abs {
+			ec.EventAfterTime(f, now.Add(d))
+		} else {
+			ec.EventAfterDuration(f, d)
+		}
+	}
+	try(true, true, time.Minute)
+	for i := 0; i < batchSize; i++ {
+		d := time.Duration(rand.Intn(30)-3) * time.Second
+		try(i%2 == 0, d >= 0, d)
+	}
+	ec.Run(nil)
+	if numDone != batchSize+1 {
+		t.Errorf("Got only %v events", numDone)
+	}
+	lastTime := now.Add(-3 * time.Second)
+	for i := 0; i <= batchSize; i++ {
+		nextTime := <-times
+		if nextTime.Before(lastTime) {
+			t.Errorf("Got %s after %s", nextTime, lastTime)
+		}
+	}
+	endTime := ec.Now()
+	dx := endTime.Sub(now)
+	if dx > time.Minute+fuzz {
+		t.Errorf("Run started at %#+v, ended at %#+v, dx=%d", now, endTime, dx)
+	}
+	now = endTime
+	var shouldRun int32
+	strictable = false
+	for i := 0; i < batchSize; i++ {
+		d := time.Duration(rand.Intn(30)-3) * time.Second
+		try(i%2 == 0, d >= 0, d)
+		if d <= 12*time.Second {
+			shouldRun++
+		}
+	}
+	ec.SetTime(now.Add(13*time.Second - 1))
+	if numDone != batchSize+1+shouldRun {
+		t.Errorf("Expected %v, but %v ran", shouldRun, numDone-batchSize-1)
+	}
+	lastTime = now.Add(-3 * time.Second)
+	for i := int32(0); i < shouldRun; i++ {
+		nextTime := <-times
+		if nextTime.Before(lastTime) {
+			t.Errorf("Got %s after %s", nextTime, lastTime)
+		}
+		lastTime = nextTime
+	}
+}
+
+// copied from baseclocktest, because apparently
+// test files can not define public symbols.
+type SettablePassiveClock interface {
+	clock.PassiveClock
+	SetTime(time.Time)
+}
+
+// copied from baseclocktest, because it is not public
+func exercisePassiveClock(t *testing.T, pc SettablePassiveClock) {
```

Comment:
Let's not introduce unnecessary interfaces for the sake of introducing them.

This function is used in exactly one place. So let's simply change argument type to `TestableEventClock` and get rid of SettablePassiveClock interface that isn't used anywhere else.

Reply:
I was going somewhere with that, but never got there and I think I am not going to, so I have removed it.